web/packages/shared/utils/semVer.ts

```diff
@@ -44,3 +44,23 @@ export const compareSemVers = (a: string, b: string): -1 | 1 | 0 => {
 
   return 0;
 };
+
+/**
+ * Extracts the major version number from a semver string.
+ * @throws Error when the input is not a valid semver string.
+ */
+export function getMajor(version: string): number {
+  const parts = version.split('.');
+  if (parts.length < 3) {
+    throw new Error(`'${version}' is not a valid semver string.`);
```

Comment:
This will reject valid semver strings such as 18.1.2-dev.ravicious.1, see https://regex101.com/r/Ly7O1x/3/

If this function is not needed in the browser, we should probably use something like the semver package instead. https://www.npmjs.com/package/semver

Reply:
Technically, `18.1.2-dev.ravicious.1` is valid, there are more than two parts in this string 🤓 

I was considering adding the `semver` package, but since I only needed a single function from it...
Plus it doesn't have builtin types, so we would need `@types/semver` too.

I know that my validation isn't too sophisticated, but I only wanted to check if the string passes the base semver requirements.
Given that the string above is parsed correctly, do you still it's worth adding the package?

Reply:
If using https://www.npmjs.com/package/semver means no need to worry about any custom version quirks ever, I would be inclined to use that. It will be 

We are already using external semver package in Go https://github.com/coreos/go-semver. 

Conversely, I am curious why can't the decision for the client to be updated/what version to upgrade/downgrade cannot be made in the server? We do we need to handle this in JS?